### PR TITLE
feat: add C implementation for `math/base/special/binomcoef`

### LINKED ISSUE #2206: [RFC]: add C implementation for `math/base/special/binomcoef`
### Description

This RFC proposes adding C implementation for [`math/base/special/binomcoef`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/binomcoef).

```C
double stdlib_base_binomcoef( const double n, const double k );
```

### Related Issues

Related issues #649.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by gunjjoshi**
Resolves #2206.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds `C` implementation for [`math/base/special/binomcoef`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/binomcoef).

```c
double stdlib_base_binomcoef( const int64_t n, const int64_t k )
```

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #2206 and a part of #649.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by gunjjoshi** [comment]
I have excluded the tests to check when the input in either `NaN` or is not an integer, as `stdlib_base_is_nan` and `stdlib_base_is_integer` require inputs of `double` type, but here we're dealing with `int64_t`.

**Comment by gunjjoshi** [comment]
I've fixed the PR title, but still the check is failing.

**Comment by kgryte** [comment]
@gunjjoshi Try merging in the latest `develop` and pushing that up. For whatever reason, GitHub is not pulling in the update title.

File: lib/node_modules/@stdlib/math/base/special/binomcoef/manifest.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      "include": [
+        "./include"
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/math/base/napi/binary",
+        "@stdlib/math/base/special/cbrt",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/binomcoef/manifest.json]: Not seeing `@stdlib/math/base/special/cbrt` being used; can you please double-check?

**Review by Planeshifter — CHANGES_REQUESTED** [review]
One minor concern, but otherwise should be ready to land!

File: lib/node_modules/@stdlib/math/base/special/binomcoef/manifest.json (hunk 1 above)
Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/binomcoef/manifest.json]: Thanks for catching this! I've removed this now.

**Review by Planeshifter — APPROVED** [review]
Let's merge after CI has passed!

**Pull request merged**